sycl/doc/extensions/proposed/sycl_ext_oneapi_launch_queries.asciidoc

```diff
@@ -0,0 +1,263 @@
+may be limited on some (highly parallel) devices. If certain properties (e.g.
+forward progress guarantees, cross-work-group synchronization) are being used
+as part of a performance optimization, developers should check that the values
+returned by these queries is not 1.
+
+[source, c++]
+----
+namespace ext::oneapi::experimental::info::kernel {
+
+template <uint32_t Dimensions>
+struct max_work_item_sizes;
+
+struct max_work_group_size;
+struct max_num_work_groups;
+
+}
+----
+
+[%header,cols="1,5,5,5"]
+|===
+|Kernel Descriptor
+|Argument Types
+|Return Type
+|Description
+
+|`template <uint32_t Dimensions>
+ max_work_item_sizes`
+|`sycl::queue`
+|`id<Dimensions>`
+|Returns the maximum number of work-items that are permitted in each dimension
+ of a work-group, when the kernel is submitted to the specified queue,
+ accounting for any kernel properties or features. If the kernel can be
+ submitted to the specified queue without an error, the minimum value returned
+ by this query is 1, otherwise it is 0.
+
+|`max_work_group_size`
+|`sycl::queue`
+|`size_t`
+|Returns the maximum number of work-items that are permitted in a work-group,
+when the kernel is submitted to the specified queue, accounting for any
+kernel properties or features. If the kernel can be submitted to the specified
+queue without an error, the minimum value returned by this query is 1,
+otherwise it is 0.
+
+|`max_num_work_groups`
+|`sycl::queue`, `sycl::range`, `size_t`
+|`size_t`
+|Returns the maximum number of work-groups, when the kernel is submitted to the
+specified queue with the specified work-group size and the specified amount of
+dynamic work-group local memory (in bytes), accounting for any kernel
+properties or features. If the kernel can be submitted to the specified queue
+without an error, the minimum value returned by this query is 1, otherwise it
+is 0.
+
+|===
+
+A separate set of launch queries can be used to reason about how an
+implementation will launch a kernel on the specified queue. The values of these
+queries should also be checked if a kernel is expected to be launched in a
+specific way (e.g., if the kernel requires two sub-groups for correctness).
+
+[source, c++]
+----
+namespace ext::oneapi::experimental::info::kernel {
+
+struct max_sub_group_size;
+struct num_sub_groups;
+
+}
+----
+
+[%header,cols="1,5,5,5"]
+|===
+|Kernel Descriptor
+|Argument Types
+|Return Type
+|Description
+
+|`max_sub_group_size`
+|`sycl::queue`, `sycl::range`
+|`uint32_t`
+|Returns the maximum sub-group size, when the kernel is submitted to the
+specified queue with the specified work-group size, accounting for any kernel
+properties or features. The return value of this query must match the value
+returned by `sub_group::get_max_local_range()` inside the kernel. If the kernel
+can be submitted to the specified queue without an error, the minimum value
+returned by this query is 1, otherwise it is 0.
+
+|`num_sub_groups`
```

Comment:
I saw that you commented about the name of this descriptor elsewhere.  By naming it "num_sub_groups" does this imply that every work-group will have exactly this many sub-groups?  Is it guaranteed that every work-group has the same number of sub-groups?

Reply:
Yes, this is guaranteed in SYCL because every work-group in a kernel has the same size.